syntax = "proto3";

package cliente;

import "google/protobuf/empty.proto";

service ClienteController {
    rpc List(ClienteListRequest) returns (stream Cliente) {}
    rpc Create(Cliente) returns (Cliente) {}
    rpc Retrieve(ClienteRetrieveRequest) returns (Cliente) {}
    rpc Update(Cliente) returns (Cliente) {}
    rpc Destroy(Cliente) returns (google.protobuf.Empty) {}
}

message Cliente {
    string cd_cliente = 1;
    string nome = 2;
    string sobrenome = 3;
    string cpf = 4;
    string sexo = 5;
}

message ClienteListRequest {
}

message ClienteRetrieveRequest {
    string cd_cliente = 1;
}
